### PR TITLE
fix(mssql): add ability to specify schema in comments

### PR DESCRIPTION
**Pull request by sykesey**
## Pull Request Checklist

- [X] Have you added new tests to prevent regressions?
- [n/a] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [n/a] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Update to the MSSQL dialect handler to handle schemas other than "dbo". At the moment comments on columns do not support schemas other than dbo as dbo is hardcoded. I have fixed this and updated the unit tests for MSSQL.

## Todos

- [ ] 
- [ ] 
- [ ]

### CODE REVIEW

File: packages/core/src/dialects/mssql/query-generator.js
Comment (ephys) [inline packages/core/src/dialects/mssql/query-generator.js:309]: you can normalize the table by using `this.extractTableDetails(table)`, it will handle things like using the proper default schema

**Review by ephys — CHANGES_REQUESTED** [review]
Otherwise this looks good, thank you! :)

File: packages/core/test/unit/dialects/mssql/query-generator.test.js
Comment (WikiRik) [inline packages/core/test/unit/dialects/mssql/query-generator.test.js:185]: Nitpicky, but we could use backticks (`) here to reduce the amount of escaping needed

File: packages/core/test/unit/dialects/mssql/query-generator.test.js
Comment (WikiRik) [inline packages/core/test/unit/dialects/mssql/query-generator.test.js:84]: Can be in a future PR due to how we're testing `createTableQuery` instead of `commentTemplate` but adding a test where we set the schema globally would be nice. For example you can see the TypeScript test files for each function in the query generator unit test folder

**Review by WikiRik — COMMENTED** [review]
Two small comments that don't necessarily need to be addressed in this PR

File: packages/core/test/unit/dialects/mssql/query-generator.test.js
Comment (sykesey) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js:84]: It looks like the test code doesn't have a huge amount of coverage right now for anything using non-standard schemas. I think you are right that this can be a future PR or get added on the to-do list. For now I'm happy to throw a couple more schema checks e.g. in create table and create column queries if need be.

File: packages/core/src/dialects/mssql/query-generator.js
Comment (sykesey) [reply packages/core/src/dialects/mssql/query-generator.js:309]: Makes sense - i've adjusted that and tested - I'll incorporate that

Comment (sykesey) [reply packages/core/src/dialects/mssql/query-generator.js]: That commit didn't work - my bad as I wasn't testing correctly locally. I've fixed it up now, hopefully the pipeline checks work now.

**Review by ephys — APPROVED** [review]
@WikiRik When it's good on your end, this is good for me too

**Review by ephys — CHANGES_REQUESTED** [review]
Shit, sorry I just noticed something

You added the test to the mssql test folder
These are legacy folders that we're working on deleting. We want tests to run on all dialects

Could you move your new & updated tests to `test/unit/query-generator/create-table-query.test.ts`? (note the ts extension)

File: packages/core/test/unit/dialects/mssql/query-generator.test.js
Comment (WikiRik) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js:84]: Yeah, it's something we're slowly adding. Let's just do it in a future PR so we can get this merged once these tests have been moved to `test/unit/query-generator/create-table-query.test.ts`

Comment (sykesey) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js]: @WikiRik @ephys I have added in the unit tests under query-generator - i had to create the create-table-query.test.ts file as I couldn't see it in the repo. I have added in the same tests as I was using but have defined them for the other dialects, so the unit tests complete. I've also reverted the extra schema test in the mssql dialect unit test.

Comment (WikiRik) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js]: Yes, I thought we already had the file but I'm creating it in https://github.com/sequelize/sequelize/pull/15526
Maybe I can clean up the unit tests there first and we merge this after? If you're fine with that

Comment (sykesey) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js]: @WikiRik Whichever is easiest, no hurry on my end.

Comment (WikiRik) [reply packages/core/test/unit/dialects/mssql/query-generator.test.js]: I've cleaned up the unit tests there, just waiting for reviews

**Review by ephys — DISMISSED** [review]

**Comment by ephys** [comment]
We'll merge this after https://github.com/sequelize/sequelize/pull/15526 but otherwise this looks good on my end

**Comment by WikiRik** [comment]
#15526 has been merged so there are some merge conflicts that need to be resolved

**Comment by sykesey** [comment]
Reopening this one - i have merged it all in

**Review by ephys — DISMISSED** [review]

File: packages/core/src/dialects/abstract/query-generator.js
Comment (WikiRik) [inline packages/core/src/dialects/abstract/query-generator.js:104]:
```suggestion
```

It's implemented in all dialects. We only need this once we have migrated it to TypeScript

**Review by WikiRik — CHANGES_REQUESTED** [review]
Can you add a test for `queryGenerator.createTableQuery({ tableName: 'myTable', schema: 'mySchema' }, { myColumn: 'DATE COMMENT Foo' })`?

Also; can you revert the changes to yarn.lock?

File: packages/core/test/unit/query-generator/create-table-query.test.ts
Comment (WikiRik) [inline packages/core/test/unit/query-generator/create-table-query.test.ts:154]: You can add the following test just before this test;
```ts
  it('produces a query to create a table with schema in tableName object and a comment', () => {
    expectsql(queryGenerator.createTableQuery({ tableName: 'myTable', schema: 'mySchema' }, { myColumn: 'DATE COMMENT Foo' }), {
      default: 'CREATE TABLE IF NOT EXISTS [mySchema].[myTable] ([myColumn] DATE COMMENT Foo);',
      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `mySchema`.`myTable` (`myColumn` DATE COMMENT Foo) ENGINE=InnoDB;',
      postgres: `CREATE TABLE IF NOT EXISTS "mySchema"."myTable" ("myColumn" DATE); COMMENT ON COLUMN "mySchema"."myTable"."myColumn" IS 'Foo';`,
      mssql: `IF OBJECT_ID(N'[mySchema].[myTable]', 'U') IS NULL CREATE TABLE [mySchema].[myTable] ([myColumn] DATE);
        EXEC sp_addextendedproperty @name = N'MS_Description', @value = N'Foo', @level0type = N'Schema', @level0name = N'mySchema',
        @level1type = N'Table', @level1name = [myTable], @level2type = N'Column', @level2name = [myColumn];`,
      sqlite: 'CREATE TABLE IF NOT EXISTS `mySchema.myTable` (`myColumn` DATE COMMENT Foo);',
      db2: `CREATE TABLE IF NOT EXISTS "mySchema"."myTable" ("myColumn" DATE); -- 'Foo', TableName = "mySchema"."myTable", ColumnName = "myColumn";`,
      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "mySchema"."myTable" ("myColumn" DATE COMMENT Foo); END`,
    });
  });
```

**Comment by WikiRik** [comment]
@sykesey a reminder for this, the test I've requested is in the comment above. So just some small tweaks and resolving of conflicts and then we can merge this PR

**Review by WikiRik — APPROVED** [review]

**Pull request merged**
